mdata/cache/if.go

```diff
@@ -6,6 +6,7 @@ import (
 
 type Cache interface {
 	Add(string, uint32, chunk.IterGen)
+	CacheIfHot(string, uint32, *chunk.IterGen)
```

Comment:
You're right that's tiny for a `struct`, but I'd guess it's still significantly bigger than a pointer, no? I can change it to passing by value anyway, shouldn't really make a big difference.